src/osdep/linux/sysinfo.cpp

```diff
@@ -97,23 +97,49 @@ static void old_Hertz_hack(void){
   seconds = (up_1 + up_2) / 2;
   h = (unsigned)( (double)jiffies/seconds/smp_num_cpus );
   /* actual values used by 2.4 kernels: 32 64 100 128 1000 1024 1200 */
-  switch(h){
-  case    9 ...   11 :  Hertz =   10; break; /* S/390 (sometimes) */
-  case   18 ...   22 :  Hertz =   20; break; /* user-mode Linux */
-  case   30 ...   34 :  Hertz =   32; break; /* ia64 emulator */
-  case   48 ...   52 :  Hertz =   50; break;
-  case   58 ...   61 :  Hertz =   60; break;
-  case   62 ...   65 :  Hertz =   64; break; /* StrongARM /Shark */
-  case   95 ...  105 :  Hertz =  100; break; /* normal Linux */
-  case  124 ...  132 :  Hertz =  128; break; /* MIPS, ARM */
-  case  195 ...  204 :  Hertz =  200; break; /* normal << 1 */
-  case  253 ...  260 :  Hertz =  256; break;
-  case  393 ...  408 :  Hertz =  400; break; /* normal << 2 */
-  case  790 ...  808 :  Hertz =  800; break; /* normal << 3 */
-  case  990 ... 1010 :  Hertz = 1000; break; /* ARM */
-  case 1015 ... 1035 :  Hertz = 1024; break; /* Alpha, ia64 */
-  case 1180 ... 1220 :  Hertz = 1200; break; /* Alpha */
-  default:
+  if        ( (h >= 9   ) && (h <= 11  ) ){
```

Comment:
This appears to be in place to support very old versions of linux.
If we're happy to support only POSIX.1-2008 and onwards, then this case can all be replace with something like:

```
 if((hz = sysconf(_SC_CLK_TCK)) > 0){
    Hertz = hz;
  }
```
which also gives an exact answer instead of an assumed convention.

That all said, your fix to silence the compiler warnings about using GNU extensions is good here.

Reply:
@evintila Please integrate this  suggestion.